common/djangoapps/student/tests/tests.py

```diff
@@ -930,3 +933,182 @@ def test_roundtrip_with_unicode_course_id(self):
+        """
+        data = dict(
+            internal_service_url="http://internal/",
+            public_service_url="http://public/",
+            api_version_number=1
+        )
+        ProgramsApiConfig(**dict(data, **kwargs)).save()
+
+    def _create_program_data(self, data):
+        """Dry method to create testing programs data."""
+        programs = {}
+        for course in data:
+            programs[unicode(course[0])] = {
+                'category': 'xseries',
+                'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                'marketing_slug': 'fake-marketing-slug-xseries-1',
+                'status': course[1],
+                'course_codes': [
+                    {
+                        'display_name': 'Demo XSeries Program 1',
+                        'key': 'TEST_A',
+                        'run_modes': [{'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'}]
+                    }
+                ],
+                'subtitle': 'Dummy program 1 for testing',
+                'name': 'First Program'
+            }
+
+        return programs
+
+    @ddt.data(
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug1'),
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}], 3, 'slug2'),
+        ('active', [], 0, ''),
+        ('inactive', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug3'),
+    )
+    @ddt.unpack
+    def test_get_xseries_programs_method(self, status, course_codes, course_codes_count, slug):
+        with patch('student.views._get_course_programs') as mock_data:
+            mock_data.return_value = {
+                unicode('course-v1:ManTestX+ManTest2+2014'): {
+                    'category': 'xseries',
+                    'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                    'marketing_slug': slug,
+                    'status': status,
+                    'course_codes': course_codes,
+                    'subtitle': 'Dummy program 1 for testing',
+                    'name': 'First Program'
+                }
+            }
+            parse_data = _get_course_programs(
+                self.user, [
+                    u'course-v1:ManTestX+ManTest2+2014', 'edx/demox/DemoCourse'
+                ]
+            )
+
+            if status == 'inactive':
+                self.assertDictEqual({}, parse_data)
+            else:
+                self.assertDictEqual(
+                    {
+                        'course-v1:ManTestX+ManTest2+2014': {
+                            'course_count': course_codes_count,
+                            'display_name': 'First Program',
+                            'program_marketing_url': urljoin(
+                             settings.MKTG_URLS.get('ROOT'), 'xseries' + '/{}').format(slug)
+                        }
+                    },
+                    parse_data
+                )
```

Comment:
i'm confused by what's being asserted in this test - it seems like we're not actually testing `_get_course_programs` because you are mocking its return value above on #L1004.   I would have expected this test to mock the return value of `get_course_programs_for_dashboard` instead, but maybe I'm missing something?